src/blocks/BoardListBlock.tsx

```diff
@@ -1,38 +1,211 @@
+
+import ActionLink from "buttons/ActionLink";
+import BoardIcon from "board/BoardIcon";
+import CircleButton from "buttons/CircleButton";
+import DefaultTheme from "theme/default";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-import DropdownMenu from "common/DropdownListMenu";
-import React from "react";
+import HighlightedText from "common/HighlightedText";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
 import { LinkWithAction } from "types";
+import React from "react";
+import classnames from "classnames";
 import { extractCompounds } from "utils/compound-utils";
+import { faArrowAltCircleRight } from "@fortawesome/free-regular-svg-icons";
 
 const Empty: React.FC<unknown> = ({ children }) => <p>{children}</p>;
 
 const Item = ({
   slug,
+  avatar,
+  color,
   description,
+  link,
   _selected,
   _onSelected,
   options,
+  updates,
+  muted,
+  outdated,
+  _onPinned,
+  _onMuted,
 }: ItemProps) => (
   <>
-    <button className="slug-container" onClick={() => _onSelected?.(slug)}>
-      !{slug}
-    </button>
-    {options.length !== 0 && (
-      <DropdownMenu options={options} zIndex={200} label="board list options">
-        <div className="board-list-options">
-          <FontAwesomeIcon icon={faEllipsisV} />
+    <div className="item-summary">
+      <ActionLink link={link}>
+        <BoardIcon
+          avatar={avatar}
+          color={color}
+          updates={updates}
+          muted={muted}
+          outdated={outdated}
+          large
+        />
+      </ActionLink>
+      <button
+        onClick={() => _onSelected?.(slug)}
+        className={classnames("slug-container", {
+          "has-updates": !!updates,
+          muted: !!muted,
+          outdated: !!outdated,
+          "with-options": !!options.length,
+        })}
+      >
+        <span className="slug">!{slug}</span>
+      </button>
+      {options.length !== 0 && (
+        <DropdownMenu
+          options={options}
+          zIndex={200}
+          label="board list options"
+          buttonClassName="insanity"
+          style={DropdownStyle.DARK}
+        >
+          <FontAwesomeIcon
+            icon={faEllipsisV}
+            size="lg"
+            color={DefaultTheme.MENU_ITEM_ICON_COLOR}
+          />
+        </DropdownMenu>
+      )}
+    </div>
+    {_selected && (
+      <div className="item-details">
+        <BoardPreview
+          slug={slug}
+          avatar={avatar}
+          description={description}
+          color={color}
+          muted={muted}
+          displayStyle={DisplayStyle.MINI}
+        />
+        <div className="board-info">
+          <HighlightedText highlightColor={color}>
+            <span className="big-slug">!{slug}</span>
+          </HighlightedText>
+
+          {description}
+          <div className="board-info-footer">
+            <div className="toggle-buttons">
+              <CircleButton
+                icon={{ icon: faThumbTack }}
```

Comment:
Since we don't have a different indicator of whether the board is pinned or not, we should make the color of this button be gray if it's not pinned and white otherwise